pymc3/model.py

```diff
@@ -1695,16 +1696,25 @@ def pandas_to_array(data):
     XXX: When `data` is a generator, this will return a Theano tensor!
 
     """
-    if hasattr(data, "values"):  # pandas
-        if data.isnull().any().any():  # missing values
-            ret = np.ma.MaskedArray(data.values, data.isnull().values)
+    if isinstance(data, (pd.Series, pd.DataFrame)):
+        mask = data.isnull()
+        if mask.any().any():
+            # there are missing values
+            ret = np.ma.MaskedArray(data.values, mask.values)
         else:
             ret = data.values
-    elif hasattr(data, "mask"):
-        if data.mask.any():
-            ret = data
-        else:  # empty mask
+    elif isinstance(data, np.ndarray):
+        if isinstance(data, np.ma.MaskedArray) and not data.mask.any():
```

Comment:
What happens when `data` is already a masked array but there _are_ missing values (i.e `data.mask.any() == True`)? IIUC, you're casting `data` again as masked array below, but shouldn't there be an `elif` clause here?
```python
elif isinstance(data, np.ma.MaskedArray) and data.mask.any():
    ret = data
```

Reply:
You're right. Thanks for catching this.